open_pull_requests.sh

```diff
@@ -1,37 +1,91 @@
-#!/usr/bin/env bash
+#!/bin/bash
+
+# Script for opening pull requests for CVEs found by cvejob.
+# cvejob stores CVEs in database/ directory.
+#
+# Requirements:
+# GITHUB_TOKEN environment variable needs to contain a GitHub token
+# which will be used to make authenticated API calls that will open
+# new pull requests in the upstream CVE database.
+# See: https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line
 
 set -e
-set -x
 
-pwd
-ls -la
+if [ -z ${GITHUB_TOKEN} ]; then
+    echo "Please provide GitHub token in GITHUB_TOKEN environment variable. Exiting..."
+    exit 1
+fi
+
+here="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+
+data=${here}/database
+mkdir -p ${data}
 
+# check if there is at least one CVE in the cvejob-local database
+file_count=`find ${data} -type f -name "*.yaml" | wc -l`
+if [[ ${file_count} -eq 0 ]]; then
+    echo "No CVEs, no PRs :)"
+    exit 0
+fi
+echo "${file_count} CVEs found in the cvejob-local database"
 
-files=$(git status -s | awk '{ print $2 }')
+# clone/update upstream cve db
+cvedb_git_url=https://github.com/fabric8-analytics/cvedb.git
+cvedb=${here}/cvedb
+echo "Cloning/updating the upstream database..."
+git clone ${cvedb_git_url} cvedb || (
+    cd ${cvedb} && \
+    git checkout master &&\
+    git reset --hard origin/master &&\
+    git clean -f -d &&\
+    git pull &&\
+    git remote set-url --push origin https://${GITHUB_TOKEN}@github.com/fabric8-analytics/cvedb.git
+)
```

Comment:
this looks a bit dangerous, but let's try it on production :)

Reply:
Why do you think so? :)